### DISCUSSION TITLE
tailwind intellisense not detecting duplicate utilities when plugin added

### DISCUSSION THREAD

**Discussion by binaryartifex** [tailwindlabs/tailwindcss#13638] [Help] [answered]
In an effort to get variable font weights working, i created a custom plugin exactly as shown [here](https://github.com/tailwindlabs/tailwindcss/discussions/3225#discussioncomment-6063402). however tailwinds intellisense still detects the core utilities of the same name...

![image](https://github.com/tailwindlabs/tailwindcss/assets/68464707/763934f4-2243-4ebc-a218-779cda519da8)

and the runtime build still seems to compile with the original utilities but just being overridden. Im not entirely sure of the implications of this, but the duplicate intellisense has be concerned there'll inevitably be some weirdness tied to some kind of weird race conditions between my font plugin and the core system. Why aren't my utilities of the same name just outright replacing the core ones?

For context, the environment is a brand new scaffolded remix app, textbook tailwind integration with the custom plugin applied via the link above.

EDIT: just did a quick build to check if this persists into the production css, it does.

![image](https://github.com/tailwindlabs/tailwindcss/assets/68464707/d8c2fac2-ef93-4afc-8c8c-7f6426f9db42)

**Comment by wongjn** [comment, marked as answer]
You coul [disable the core `fontWeight` plugin](https://tailwindcss.com/docs/configuration#core-plugins) so that only the CSS from the plugin is built:

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  corePlugins: {
    fontWeight: false,
  }
}
```

**Reply by binaryartifex** [reply]
ahh yep, this seems to work, cheers. seems kinda odd though that it doesn't do a like for like replacement...clearly the intent is to replace the original values
